### DISCUSSION TITLE
Querying content items by taxonomy

### DISCUSSION THREAD

**Discussion by milosh-96** [OrchardCMS/OrchardCore#16155] [General]
So I'm working on a my site and I have a glossary section. I would like to query content items by taxonomy (topic) and to put results in these boxes. So each box should contain terms of one topic (currently it's hard coded of course).

What would be an ideal way to do that? Or maybe there is a module? 

For now I would manually enter IDs where it's needed, but developing a widget would be better of course and it's definitely a TODO. My main issue is like - how to even get results. For example, in Wordpress there would be that $posts = new WP_Query($args) so I'm looking for something like that. I know there are SQL Queries in the admin panel, but I'm not sure it's the right way.

![image](https://github.com/OrchardCMS/OrchardCore/assets/13017737/82723f08-a6f1-4ec7-99a1-ec666605f579)

**Comment by hishamco** [comment]
What about using `QueryCategorizedContentItemsAsync()`

https://github.com/OrchardCMS/OrchardCore/blob/cab8dbd7ae6830b6eb066c7279c647023615e59d/src/OrchardCore.Modules/OrchardCore.Taxonomies/Helper/TaxonomyOrchardHelperExtensions.cs#L63-L71

**Reply by milosh-96** [reply]
So I wrote an ad-hoc shape that takes parameters so I can just add them to the home page template.

![image](https://github.com/OrchardCMS/OrchardCore/assets/13017737/745f2fdf-ec48-439f-9971-ae77d3cbab32)

This is not the final solution, but it works so I can continue with other stuff.

**Comment by milosh-96** [comment]
Thanks! The method is very useful for this.

**Reply by hishamco** [reply]
Happy to hear that
